lib/ddtrace/contrib/http/patcher.rb

```diff
@@ -18,7 +18,7 @@ def should_skip_tracing?(req, address, port, transport, pin)
         # we don't want to trace our own call to the API (they use net/http)
         # when we know the host & port (from the URI) we use it, else (most-likely
         # called with a block) rely on the URL at the end.
-        if req.uri
+        if req.respond_to?(:uri) && req.uri
```

Comment:
Some people like to audit/log all requests, so they converge on something like faraday/patreon to provide these hooks.

It's a fairly minor issue until someone asks for it.